elb.tf

```diff
@@ -0,0 +1,81 @@
+resource "aws_security_group" "elb" {
+  name_prefix = "${var.name}-elb-sg"
+  description = "Inbound ELB"
+  vpc_id      = var.vpc_id
+
+  tags = merge(
+    var.tags,
+    var.elb_additional_sg_tags,
+    { "Name" : "${var.name}-elb-sg" }
+  )
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_security_group_rule" "elb_egress" {
+  description              = "Allow traffic from the ELB to the instances"
+  from_port                = 80
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.elb.id
+  source_security_group_id = aws_security_group.this.id
+  to_port                  = 80
+  type                     = "egress"
+}
+
+resource "aws_security_group_rule" "elb_ingress" {
+  count = length(var.elb_allowed_cidr_blocks) > 0 ? 1 : 0
+
+  cidr_blocks       = var.elb_allowed_cidr_blocks #tfsec:ignore:AWS006
+  description       = "Allow traffic from the allowed ranges"
+  from_port         = 443
+  protocol          = "tcp"
+  security_group_id = aws_security_group.elb.id
+  to_port           = 443
+  type              = "ingress"
+}
+
+resource "aws_lb" "this" {
+  name_prefix                      = substr(var.name, 0, 6)
+  enable_cross_zone_load_balancing = "true"
+  internal                         = var.elb_internal
+  load_balancer_type               = "application"
+  security_groups                  = [aws_security_group.elb.id]
+  subnets                          = var.elb_subnets
+  tags                             = var.tags
+}
+
+resource "aws_lb_listener" "this" {
+  certificate_arn   = var.elb_certificate
+  load_balancer_arn = aws_lb.this.id
+  port              = 443
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-TLS-1-2-2017-01"
```

Comment:
Do you really want to hard code this?

Also are you sure we don't want `ELBSecurityPolicy-FS-1-2-Res-2019-08`?

Reply:
good catch. I decided against the restrictive cipher set as a default but did move to FS-1-2-2019-08. I  had no idea these cipher sets existed.. the AWS documentation splits the cipher sets into two tables and puts the oldest ciphers first.

Reply:
Yeah, the AWS docs lean towards compatibility rather than security.